code/modules/events/solarflare.dm

```diff
@@ -0,0 +1,29 @@
+/datum/event/solar_flare
+	startWhen = 10
+	endWhen = 500 // = ~500x2 seconds or ~16m40s
+
+/datum/event/solar_flare/announce()
+	GLOB.event_announcement.Announce("A solar flare has been detected on collision course with the station.", "Flare Alert", new_sound = 'sound/AI/attention.ogg')
+
+/datum/event/solar_flare/start()
+	// Solars produce 40x as much power. 240KW becomes 9.6MW. Enough to cause APCs to arc all over the station if >=2 solars are hotwired.
```

Comment:
Part of me wants to add an orange tint to space tiles, in the same way that they glow green when theres a radstorm, but make it much less saturated, so its noticeable but not intrusive

Reply:
I would, but I don't see any way to do that.

Reply:
Is it possible to take from the Ashstorm code at all to make it affect all non-room marked areas on the Z-Level? When Elysian does the Lavaland Refugee event, we have to use Permits and such to mark rooms so they dont get burned, so perhaps its possible to use the same system for the tint?

Reply:
I did some testing making a weather event called firestorm. Weather events let you define what areas get overlays. So its a simple matter of making an icon and setting the affected area to /turf/space.

Reply:
Converted to weather datum.